promql/functions.go

```diff
@@ -513,6 +513,15 @@ func funcAbsentOverTime(vals []parser.Value, args parser.Expressions, enh *EvalN
 		})
 }
 
+// === present_over_time(Vector parser.ValueTypeMatrix) Vector ===
+// This function will return 1 if the matrix has at least one element.
+// Due to engine optimization, this function is only called when this condition is true.
```

Comment:
I believe this copied comment only applies to `absent_over_time()`, for which this engine optimization exists: https://github.com/prometheus/prometheus/blob/7337ecf0d353d41fed2c1352833862a02f73e18f/promql/engine.go#L1366-L1402

Reply:
Hmm, I thought it was appropriate since both the functions are called only when the corresponding data exists. 
https://github.com/prometheus/prometheus/blob/b1ed4a0a663d0c62526312311c7529471abbc565/promql/engine.go#L1343

Removed...

Reply:
Ah yeah, you were right, sorry. Still fine to omit that line I think, it's normal for any `_over_time()` function to only be called when there's data for it :)

For `absent_over_time()` it probably just made more sense to call it out because the function returns something even (and only if) it was *not* normally called.